### PR TITLE
Universal instance type

### PR DESCRIPTION
**Pull request by nmenardg**
https://www.pivotaltracker.com/story/show/126630419
https://www.pivotaltracker.com/story/show/128009829

### CODE REVIEW

**Comment by oleduc** [comment]
Approved.

File: nefertari/events.py [hunk 1]
```diff
@@ -298,14 +300,14 @@ def trigger_events(view_obj):
         event_kwargs = {
             'view': view_obj,
             'model': view_obj.Model,
+            'initial_state': view_obj.initial_state,
```
Comment (nikitagromov) [inline nefertari/events.py]: initial_state is field of Request instance, how we can get it from view_obj ?

**Comment by nikitagromov** [comment]
Add unit-tests, please

**Comment by oleduc** [comment]
@nmenardg We should add a test for RequestEvent and BaseView constructors to make sure they specify a initial_state. Also we should add a test to make sure event.instance is populated when view_obj.context is properly set. But then again, this last one feels like a change detector.

**Comment by nikitagromov** [comment]
Approved

File: tests/test_events.py [hunk 2]
```diff
@@ -1,17 +1,18 @@
 from mock import patch, Mock, call
 
 from nefertari import events
+from nefertari_sqla import ESBaseDocumengit commit t
```
Comment (nikitagromov) [inline tests/test_events.py]: @nmenardg This is typo ?)

**Comment by nikitagromov** [comment]
Waiting for fix typo and then Approved

File: tests/test_events.py (hunk 2 above)
Comment (nmenardg) [inline tests/test_events.py]: XD

**Comment by nikitagromov** [comment]
Approved

**Pull request merged**